src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -171,11 +171,16 @@ public void start(final Map<String, String> parsedConfig) {
         Boolean.parseBoolean(parsedConfig.get(SnowflakeSinkConnectorConfig.REBALANCING));
 
     // default to snowpipe
+    // If it is snowpipe_streaming, set delivery guarantee to exactly once.
     IngestionMethodConfig ingestionType = IngestionMethodConfig.SNOWPIPE;
     if (parsedConfig.containsKey(SnowflakeSinkConnectorConfig.INGESTION_METHOD_OPT)) {
```

Comment:
Can the user set this? If they can and they set the delivery guarantee to anything other than exactly once we should error out.

Reply:
Correct, we will err out if they mistakenly set this as atleast once